Add Chart.Residual – Modified Lollipop Chart Implementation

LibraChris opened the pull request:
## Description:
This PR adds Chart.Residual, a customized variant of the Lollipop chart (see #417) tailored for showcasing deviation of values to a reference values. 

## Key updates:
- Add Chart.Residual to _ChartAPI/Chart2D_
- Add Basic documentation and usage examples under _DISTRIBUTION CHARTS_.

LibraChris committed:
add Residue Chart

LibraChris committed:
update Residual chart arguments

LibraChris committed:
Add Documentation for Residual Chart

kMutagene commented on docs/distribution-charts/residual-chart.fsx:
rename to just "residual.fsx", as the path will then be `/distribution-charts/residual.html` on the hosted docs

kMutagene replied:
also, add it to the `04_distribution-charts` solution folder

kMutagene commented on src/Plotly.NET/ChartAPI/Chart2D.fs:
```diff
@@ -6014,4 +6014,530 @@ module Chart2D =
+                )
+                |> GenericChart.mapTrace (
+                    Trace2DStyle.Scatter(
+                        LegendGroup = (defaultArg GroupName "Datapoints"),
```
To clarify from comment above, just pass `?LegendGroup = GroupName`, so the userchoice is respected, but no default is set. Same for any other legend group settings.

kMutagene commented on src/Plotly.NET/ChartAPI/Chart2D.fs:
```diff
@@ -6014,4 +6014,530 @@ module Chart2D =
+                reference: seq<#IConvertible>,
+                ?Name: string,
+                ?ReferenceName: string,
+                ?GroupName: string,
```
I think grouping the legend items here is unnecessary, as we will have one trace per group per default, and also same name for reference line and group:

<img width="231" height="157" alt="grafik" src="https://github.com/user-attachments/assets/33b940ba-3547-431c-b86d-ec4af00e1e48" />

please refactor so that, if no group names are set by the user, no trace grouping is done.

kMutagene reviewed (changes requested):
Looks great, but please do not set any defaults for legend grouping, see comments

LibraChris committed:
Rename to residual docs and add to solution folder

- Renamed docs "residual-charts.fsx" to "residual.fsx"
- Added residual.fsx to the 04_distribution-charts solution folder

LibraChris committed:
Update Legend grouping for Residual charts

- Avoid default grouping when legend groups are not given

LibraChris commented:
@kMutagene I addressed the requested changes, please have a look.

kMutagene commented on src/Plotly.NET/ChartAPI/Chart2D.fs:
```diff
@@ -6184,8 +6186,8 @@ module Chart2D =
-                        LegendGroup = (defaultArg GroupName "Reference Line"),
-                        LegendGroupTitle = (Title.init (Text = (defaultArg GroupName "Reference Line")))
+                        ?LegendGroup        = LegendGroupReference,
+                        ?LegendGroupTitle   = if LegendGroupReference.IsSome then Some (Title.init (Text = LegendGroupReference.Value)) else None
```
Either make the title a separate argument or leave it out here. this still injects defaults not visible to the user, and might actually clash if i want to assign an already existing legendgroup that has a different title.

LibraChris replied:
@kMutagene  Removed the LegendGroupTitle

LibraChris committed:
Update Residual Legend options

Remove LegendGroupTitle to avoid unintended default injection
